core/templates/dev/head/domain/topic/topic-update.service.ts

```diff
@@ -124,6 +126,44 @@ angular.module('oppia').factory('TopicUpdateService', [
           });
       },
 
+      /**
+       * Changes the abbreviated name of a topic and records the change in the
+       * undo/redo service.
+       */
+      setAbbreviatedTopicName: function(topic, abbreviatedName) {
+        var oldAbbreviatedName = angular.copy(topic.getAbbreviatedName());
+        _applyTopicPropertyChange(
+          topic, TOPIC_PROPERTY_ABBREVIATED_NAME,
+          abbreviatedName, oldAbbreviatedName,
+          function(changeDict, topic) {
+            // Apply
+            var name = _getNewPropertyValueFromChangeDict(changeDict);
+            topic.setAbbreviatedName(name);
+          }, function(changeDict, topic) {
+            // Undo.
+            topic.setAbbreviatedName(oldAbbreviatedName);
+          });
+      },
+
+      /**
+       * Changes the thumbnail data url of a topic and records the change in the
+       * undo/redo service.
+       */
+      setThumbnail: function(topic, thumbnailName) {
+        var oldThumbnailName = angular.copy(topic.getThumbnail());
```

Comment:
Again, be careful with the terminology -- distinguish between "thumbnail" and "thumbnail filename". Also I'm not sure "thumbnail name" is a thing...

Reply:
done,